pkg/core/storage/memory_store.go

```diff
@@ -85,14 +85,19 @@ func (s *MemoryStore) Delete(key []byte) error {
 // PutBatch implements the Store interface. Never returns an error.
 func (s *MemoryStore) PutBatch(batch Batch) error {
 	b := batch.(*MemoryBatch)
+	return s.PutChangeSet(b.mem, b.del)
+}
+
+// PutChangeSet implements the Store interface. Never returns an error.
+func (s *MemoryStore) PutChangeSet(puts map[string][]byte, dels map[string]bool) error {
 	s.mut.Lock()
-	defer s.mut.Unlock()
-	for k := range b.del {
-		s.drop(k)
+	for k := range puts {
+		s.put(k, puts[k])
 	}
-	for k, v := range b.mem {
-		s.put(k, v)
+	for k := range dels {
```

Comment:
It was `del` first and `put` after that before. Now it looks like any other storage (`put` first, then `del`), but the order of operations was changed. Is it important? Consider an example where the same key is both in `puts` and `dels` maps.

Reply:
I'd say that `puts` and `dels` must not have intersections and for our use cases they don't.